packages/web-worker/src/create.ts

```diff
@@ -14,9 +15,9 @@ export function getEndpoint(caller: Endpoint<any>['call']) {
   return workerEndpointCache.get(caller);
 }
 
-export function createWorker<T extends {[key: string]: () => Promise<any>}>(
+export function createWorker<T extends {[K in keyof T]: T[K]}>(
   script: () => Promise<T>,
-) {
+): () => {[K in keyof T]: PromisifyExport<T[K]>} {
```

Comment:
Hmm, I'd like to try and just rely on the inferred type of `create()` if possible. Can `create` just return `Endpoint<T>['call']` still, and we update `Endpoint` to `PromisifyModule<T>`?

Reply:
This was actually how I approached it first, by having `Endpoint['call']` infer as `PromisifyModule<T>`. When I tested it out in web-proving-ground, I got `Property '[some_function_name]' does not exist on type 'PromisifyModule<[some_module_path]>`